[Asset Inventory] Initializes generic entities with 26 hours lookback period

opauloh opened the pull request:
## Summary

Due to a recent Entity Store performance improvement ([here](https://github.com/elastic/kibana/pull/219825)) related to the large amount of initial data in the host and user entity stores, the default entity store lookback period was reduced to 3 hours. While it's a positive change, it conflicts with many asset/entities retrieval integrations that have a default sync time of 24 hours leading to several documents not appearing during the Asset Inventory onboarding since Generic Entity stores have a lookback of 3 hours.

This PR ensures the generic entities are initialized with a 26-hour lookback period during Asset Inventory onboarding to account for the fact that entity extraction integrations have a default sync window time of 24 hours and we want to cover the ingest window time with a buffer of 2 hours.

This will allow for a smoother onboarding experience in Asset Inventory for users with existing entity data derived from Entity/Asset extraction integrations (Cloud Asset Discovery for example).

## Screenshot

Generic entities are initialized with a lookback period of 26 hours

<img width="1926" height="1610" alt="image" src="https://github.com/user-attachments/assets/5794ddef-9d86-4ddc-8fc8-0956a092efc3" />

Host and user entities are initialized with the default lookback period (3 hours)

<img width="1943" height="1624" alt="image" src="https://github.com/user-attachments/assets/8bdc8d10-88ee-4bf4-a965-887a7ae779d5" />

opauloh committed:
Initializes generic entities with higher lookback

Ensures generic entities are initialized with a 26-hour lookback period during Asset Inventory onboarding.

This accounts for the default 24-hour ingest window of entity extraction integrations, providing a 2-hour buffer.

opauloh committed:
Moves lookback period explanation to constant

Moves the explanation for the lookback period for asset inventory generic entities to the constants file.

This clarifies the purpose of the constant and keeps the code cleaner.

opauloh committed:
Fixes data view name formatting

Ensures consistent formatting of data view names by removing an unnecessary space. This prevents potential issues with data view identification and management.

opauloh committed:
Adds asset inventory enable tests

Adds tests for the asset inventory enable function.

These tests cover scenarios such as enabling the entity store and generic engine when the entity store is not installed, initializing the generic engine when it is missing, and handling data view installation errors.

It also verifies that custom request body overrides are preserved while adding the lookback period.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/320653)
* Commit: b316fd9d62b606fc72c5289ce85e5c22eae88ff1

### Failed CI Steps
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/320653#01981ad7-0c8c-45a1-9fc8-53416ae4fd82)
* [Jest Tests #6](https://buildkite.com/elastic/kibana-pull-request/builds/320653#01981afe-e33d-4f0b-897d-15a1d95373b4)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/320653#01981ad7-0c8c-45a1-9fc8-53416ae4fd82) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/320653/jobs/01981ad7-0c8c-45a1-9fc8-53416ae4fd82/artifacts/01981afe-7ec7-4973-9ca6-ebad17c574d4) Jest Tests #<span></span>6 / AssetInventoryDataClient installAssetInventoryDataView function should install data view when it does not exist (404 error)
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/320653#01981afe-e33d-4f0b-897d-15a1d95373b4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/320653/jobs/01981afe-e33d-4f0b-897d-15a1d95373b4/artifacts/01981b0c-2ee7-47e0-933c-1830fc25785d) Jest Tests #<span></span>6 / AssetInventoryDataClient installAssetInventoryDataView function should install data view when it does not exist (404 error)
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/320653#01981ad7-0c8c-45a1-9fc8-53416ae4fd82) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/320653/jobs/01981ad7-0c8c-45a1-9fc8-53416ae4fd82/artifacts/01981afe-7ec8-4b25-ae2c-c83bafba3ecf) Jest Tests #<span></span>6 / AssetInventoryDataClient installAssetInventoryDataView function should work with different space IDs
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/320653#01981afe-e33d-4f0b-897d-15a1d95373b4) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/320653/jobs/01981afe-e33d-4f0b-897d-15a1d95373b4/artifacts/01981b0c-2ee8-445c-9b24-40549182b3bc) Jest Tests #<span></span>6 / AssetInventoryDataClient installAssetInventoryDataView function should work with different space IDs

cc @opauloh

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/320653","number":320653,"commit":"b316fd9d62b606fc72c5289ce85e5c22eae88ff1"}],"number":320653}
buildkite-pr-comment-->

opauloh committed:
Fixes whitespace in tests

Removes trailing whitespace from asset inventory data view names in tests.

This ensures consistency and avoids potential issues related to string comparisons or display inconsistencies.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/320954)
* Commit: f4c331654153808ee27ea8e59b79909a7cb0b71e

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

### History
* :broken_heart: [Build #320653](https://buildkite.com/elastic/kibana-pull-request/builds/320653) failed b316fd9d62b606fc72c5289ce85e5c22eae88ff1

cc @opauloh

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/320954","number":320954,"commit":"f4c331654153808ee27ea8e59b79909a7cb0b71e"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/320653","number":320653,"commit":"b316fd9d62b606fc72c5289ce85e5c22eae88ff1"}],"number":320954}
buildkite-pr-comment-->

kubasobon commented:
Verified 

<img width="3024" height="1666" alt="image" src="https://github.com/user-attachments/assets/39961b47-767c-486e-a15f-25f5b5669577" />

<img width="370" height="420" alt="Screenshot 2025-10-03 at 16 19 16" src="https://github.com/user-attachments/assets/092b1c90-72ee-4013-8930-3fe6617db15e" />

<img width="386" height="316" alt="Screenshot 2025-10-03 at 16 19 05" src="https://github.com/user-attachments/assets/056980f7-e9ee-4005-8f85-3aa32a7d1510" />
